src/cms/views/pages/page_actions.py

```diff
@@ -620,7 +620,7 @@ def grant_page_permission_ajax(request):
         "pages/_page_permission_table.html",
         {
             "page": page,
-            "page_form": PageForm(instance=page),
+            "page_form": PageForm(instance=page, region=page.region),
```

Comment:
@melegiul just for your information: This line also fixes the bug with the broken page-specific permissions.